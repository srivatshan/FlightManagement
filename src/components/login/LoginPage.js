import React from "react";
import { getCredentials } from "../../api/loginApi";
import Cookies from "universal-cookie";
import "../../../node_modules/bootstrap/dist/css/bootstrap.min.css";
import "./App.css";
import { useHistory } from "react-router-dom";
class LoginPage extends React.Component {
  state = {
    error: "",
    user: {
      userName: "",
      password: "",
    },
  };

  UserChangeHandler = (event) => {
    this.setState({
      user: Object.assign({}, this.state.user, {
        [event.target.name]: event.target.value,
      }),
    });
  };

  formSubmit = async (event) => {
    event.preventDefault();
    const res = await getCredentials(
      this.state.user.userName,
      this.state.user.password
    );
    console.log("result is ", res);
    if (res.length > 0) {
      this.setState({
        user: Object.assign({}, this.state.user, {
          userName: res[0].username,
          password: res[0].password,
        }),
      });
      const cookies = new Cookies();
      cookies.set("user", res, { path: "/" });
      this.props.history.push(`/`);
    } else {
      this.setState({ error: "Invalid UserName and Password" });
    }
  };

  render() {
    return (
      <div>
        <br />
        <br />
        <br />
        <br />

        <div className="auth-wrapper" style={{ position: "fixed" }}>
          <div className="auth-inner">
            <form onSubmit={this.formSubmit}>
              <h3>Sign In</h3>
              {this.state.error != "" && (
                <h10 style={{ color: "red" }}>{this.state.error}</h10>
              )}
              <div className="form-group">
                <label>Username</label>
                <input
                  type="text"
                  className="form-control"
                  placeholder="UserName"
                  name="userName"
                  value={this.state.user.userName}
                  onChange={this.UserChangeHandler}
                />
              </div>

              <div className="form-group">
                <label>Password</label>
                <input
                  type="password"
                  className="form-control"
                  placeholder="Enter password"
                  name="password"
                  value={this.state.user.password}
                  onChange={this.UserChangeHandler}
                />
              </div>

              <button type="submit" className="btn btn-primary btn-block">
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default LoginPage;
