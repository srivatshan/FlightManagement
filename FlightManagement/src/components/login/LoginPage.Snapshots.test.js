import React from "react";
import LoginPage from "./LoginPage";
import renderer from "react-test-renderer";

it("test login page", () => {
  const tree = renderer.create(<LoginPage />);
  expect(tree).toMatchSnapshot();
});
