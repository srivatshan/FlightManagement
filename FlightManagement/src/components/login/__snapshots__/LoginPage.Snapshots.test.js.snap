// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login page 1`] = `
<div>
  <br />
  <br />
  <br />
  <br />
  <div
    className="auth-wrapper"
    style={
      Object {
        "position": "fixed",
      }
    }
  >
    <div
      className="auth-inner"
    >
      <form
        onSubmit={[Function]}
      >
        <h3>
          Sign In
        </h3>
        <div
          className="form-group"
        >
          <label>
            Username
          </label>
          <input
            className="form-control"
            name="userName"
            onChange={[Function]}
            placeholder="UserName"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label>
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Enter password"
            type="password"
            value=""
          />
        </div>
        <button
          className="btn btn-primary btn-block"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
